### ISSUE TITLE
On JDK 21, hangs on startup

### ISSUE THREAD

**Issue by Marcus-Rosti** [http4s/http4s-netty#615]
```
11:32:19.481 [io-compute-blocker-5] DEBUG i.n.u.i.l.InternalLoggerFactory - Using SLF4J as the default logging framework
11:32:19.489 [io-compute-blocker-5] DEBUG i.n.u.i.PlatformDependent0 - -Dio.netty.noUnsafe: false
11:32:19.489 [io-compute-blocker-5] DEBUG i.n.u.i.PlatformDependent0 - Java version: 21
11:32:19.489 [io-compute-blocker-5] DEBUG i.n.u.i.PlatformDependent0 - sun.misc.Unsafe.theUnsafe: available
11:32:19.490 [io-compute-blocker-5] DEBUG i.n.u.i.PlatformDependent0 - sun.misc.Unsafe.copyMemory: available
11:32:19.490 [io-compute-blocker-5] DEBUG i.n.u.i.PlatformDependent0 - sun.misc.Unsafe.storeFence: available
11:32:19.490 [io-compute-blocker-5] DEBUG i.n.u.i.PlatformDependent0 - java.nio.Buffer.address: available
11:32:19.490 [io-compute-blocker-5] DEBUG i.n.u.i.PlatformDependent0 - direct buffer constructor: unavailable
java.lang.UnsupportedOperationException: Reflective setAccessible(true) disabled
	at io.netty.util.internal.ReflectionUtil.trySetAccessible(ReflectionUtil.java:31)
	at io.netty.util.internal.PlatformDependent0$5.run(PlatformDependent0.java:293)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:319)
	at io.netty.util.internal.PlatformDependent0.<clinit>(PlatformDependent0.java:286)
	at io.netty.util.internal.PlatformDependent.isAndroid(PlatformDependent.java:331)
	at io.netty.util.internal.PlatformDependent.<clinit>(PlatformDependent.java:86)
	at io.netty.util.internal.ClassInitializerUtil.tryLoadClasses(ClassInitializerUtil.java:32)
	at io.netty.channel.kqueue.Native.<clinit>(Native.java:65)
	at io.netty.channel.kqueue.KQueue.<clinit>(KQueue.java:38)
	at org.http4s.netty.server.NettyServerBuilder.selectNative(NettyServerBuilder.scala:150)
	at org.http4s.netty.server.NettyServerBuilder.getEventLoop(NettyServerBuilder.scala:128)
	at org.http4s.netty.server.NettyServerBuilder.bind(NettyServerBuilder.scala:253)
	at org.http4s.netty.server.NettyServerBuilder.resource$$anonfun$1$$anonfun$2$$anonfun$1(NettyServerBuilder.scala:295)
	at cats.effect.IOFiber.runLoop(IOFiber.scala:343)
	at cats.effect.IOFiber.execR(IOFiber.scala:1366)
	at cats.effect.IOFiber.run(IOFiber.scala:112)
	at cats.effect.unsafe.WorkerThread.run(WorkerThread.scala:702)
11:32:19.491 [io-compute-blocker-5] DEBUG i.n.u.i.PlatformDependent0 - java.nio.Bits.unaligned: available, true
11:32:19.491 [io-compute-blocker-5] DEBUG i.n.u.i.PlatformDependent0 - jdk.internal.misc.Unsafe.allocateUninitializedArray(int): unavailable
java.lang.IllegalAccessException: class io.netty.util.internal.PlatformDependent0$7 cannot access class jdk.internal.misc.Unsafe (in module java.base) because module java.base does not export jdk.internal.misc to unnamed module @4d5804c6
	at java.base/jdk.internal.reflect.Reflection.newIllegalAccessException(Reflection.java:394)
	at java.base/java.lang.reflect.AccessibleObject.checkAccess(AccessibleObject.java:714)
	at java.base/java.lang.reflect.Method.invoke(Method.java:571)
	at io.netty.util.internal.PlatformDependent0$7.run(PlatformDependent0.java:429)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:319)
	at io.netty.util.internal.PlatformDependent0.<clinit>(PlatformDependent0.java:420)
	at io.netty.util.internal.PlatformDependent.isAndroid(PlatformDependent.java:331)
	at io.netty.util.internal.PlatformDependent.<clinit>(PlatformDependent.java:86)
	at io.netty.util.internal.ClassInitializerUtil.tryLoadClasses(ClassInitializerUtil.java:32)
	at io.netty.channel.kqueue.Native.<clinit>(Native.java:65)
	at io.netty.channel.kqueue.KQueue.<clinit>(KQueue.java:38)
	at org.http4s.netty.server.NettyServerBuilder.selectNative(NettyServerBuilder.scala:150)
	at org.http4s.netty.server.NettyServerBuilder.getEventLoop(NettyServerBuilder.scala:128)
	at org.http4s.netty.server.NettyServerBuilder.bind(NettyServerBuilder.scala:253)
	at org.http4s.netty.server.NettyServerBuilder.resource$$anonfun$1$$anonfun$2$$anonfun$1(NettyServerBuilder.scala:295)
	at cats.effect.IOFiber.runLoop(IOFiber.scala:343)
	at cats.effect.IOFiber.execR(IOFiber.scala:1366)
	at cats.effect.IOFiber.run(IOFiber.scala:112)
	at cats.effect.unsafe.WorkerThread.run(WorkerThread.scala:702)
11:32:19.494 [io-compute-blocker-5] DEBUG i.n.u.i.PlatformDependent0 - java.nio.DirectByteBuffer.<init>(long, {int,long}): unavailable
11:32:19.494 [io-compute-blocker-5] DEBUG i.n.u.i.PlatformDependent - sun.misc.Unsafe: available
11:32:19.494 [io-compute-blocker-5] DEBUG i.n.u.i.PlatformDependent - -Dio.netty.tmpdir: /var/folders/gd/4mv4d8q553g790vd9scjd0pw0000gn/T (java.io.tmpdir)
11:32:19.494 [io-compute-blocker-5] DEBUG i.n.u.i.PlatformDependent - -Dio.netty.bitMode: 64 (sun.arch.data.model)
11:32:19.495 [io-compute-blocker-5] DEBUG i.n.u.i.PlatformDependent - Platform: MacOS
11:32:19.495 [io-compute-blocker-5] DEBUG i.n.u.i.PlatformDependent - -Dio.netty.maxDirectMemory: -1 bytes
11:32:19.495 [io-compute-blocker-5] DEBUG i.n.u.i.PlatformDependent - -Dio.netty.uninitializedArrayAllocationThreshold: -1
11:32:19.495 [io-compute-blocker-5] DEBUG i.n.u.i.CleanerJava9 - java.nio.ByteBuffer.cleaner(): available
11:32:19.495 [io-compute-blocker-5] DEBUG i.n.u.i.PlatformDependent - -Dio.netty.noPreferDirect: false
11:32:19.497 [io-compute-blocker-5] DEBUG i.n.u.i.NativeLibraryLoader - -Dio.netty.native.workdir: /var/folders/gd/4mv4d8q553g790vd9scjd0pw0000gn/T (io.netty.tmpdir)
11:32:19.497 [io-compute-blocker-5] DEBUG i.n.u.i.NativeLibraryLoader - -Dio.netty.native.deleteLibAfterLoading: true
11:32:19.497 [io-compute-blocker-5] DEBUG i.n.u.i.NativeLibraryLoader - -Dio.netty.native.tryPatchShadedId: true
11:32:19.497 [io-compute-blocker-5] DEBUG i.n.u.i.NativeLibraryLoader - -Dio.netty.native.detectNativeLibraryDuplicates: true
```

it just hangs at that point and doesn't even respond to sigterm -- any thoughts?

**Comment by hamnis** [comment]
Are you on Apple Silicon?

Can you try to see if this also happens when using NIO transport?

**Comment by hamnis** [comment]
Should be fixed by #614 which is available in `0.5.13`, please report back if this is not the case.

**Comment by Marcus-Rosti** [comment]
fixed! thanks @hamnis
